### ISSUE TITLE
VisualizationDataRecord with a regression task?

### ISSUE THREAD

**Issue by RylanSchaeffer** [pytorch/captum#784]
I'm fine-tuning BERT (using the Transformers library) to perform a regression task and I can successfully extract the attributions. However, how am I supposed to visualize the importance of different words using `VisualizationDataRecord`? 

Its arguments seem to require at least three "classes" in the context of classification and a pred_prob, none of which I have.

```
        word_attributions,
        pred_prob,
        pred_class,
        true_class,
        attr_class,
        attr_score,
        raw_input,
        convergence_score,
```

**Comment by RylanSchaeffer** [comment]
As an aside, if the `VisualizationDataRecord` is designed for classification, shouldn't it be called `ClassificationVisualizationDataRecord`?

**Comment by RylanSchaeffer** [comment]
The regression tutorial (https://captum.ai/tutorials/House_Prices_Regression_Interpret) doesn't seem to use the VisualizationDataRecord. Is there another regression tutorial which does?

**Comment by NarineK** [comment]
VisualizationDataRecord is only for text, @RylanSchaeffer ! I hope that  you were able to visualize it with custom visualization  UI. Can we close this issue ?

**Comment by RylanSchaeffer** [comment]
I am trying to visualize text. I think you might have misunderstood my question

**Comment by craymichael** [comment]
Apologies for the delay here. You are right that `VisualizationDataRecord` / `visualize_text` is shaped around classification examples, even though the token heatmap itself can be used for scalar regression outputs.

For a text regression model, attribute the scalar regression output directly. If the model returns one scalar per example, you usually do not need to pass `target` to the attribution method. The current workaround is to pass descriptive strings into the label fields and use a placeholder probability:

```python
record = visualization.VisualizationDataRecord(
    word_attributions=token_attributions,
    pred_prob=1.0,  # placeholder; visualize_text displays this as "(1.00)"
    pred_class=f"prediction={prediction:.3f}",
    true_class=f"target={label:.3f}",
    attr_class="regression output",
    attr_score=float(token_attributions.sum()),
    raw_input_ids=tokens,
    convergence_score=float(delta),
)

visualization.visualize_text([record])
```

Here, green tokens increase the scalar output relative to the baseline, red tokens decrease it, and `attr_score` is the sum of token attributions.

This is not ideal because the rendered table still has classification-oriented labels and a probability field, but it is the supported workaround with the current API. Since this should unblock regression text visualization, I'll close this for now.
